homeassistant/components/incharge/sensor.py

```diff
@@ -0,0 +1,136 @@
+        name="Total Energy Consumption",
+        suggested_display_precision=2,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the InCharge sensor."""
+    # Add entities for each charging station
+    incharge_data_coordinator = hass.data.setdefault(DOMAIN, {})[config_entry.entry_id]
+    response = await hass.async_add_executor_job(
+        incharge_data_coordinator.incharge_api.get_stations
+    )
+    for station in response.json()["stations"]:
+        entities = [
+            InChargeEntity(
+                coordinator=incharge_data_coordinator,
+                station_id=station["name"],
+                unique_id=f"{station['name']}-{description.name}",
+                description=description,
+            )
+            for _, description in SENSOR_TYPES.items()
+        ]
+        async_add_entities(entities, update_before_add=True)
+
+
+class InChargeEntity(CoordinatorEntity, SensorEntity):
+    """Representation of a InCharge Sensor."""
+
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self, coordinator, station_id, unique_id, description: SensorEntityDescription
+    ) -> None:
+        """Initialize a InCharge sensor."""
+        super().__init__(coordinator, context=station_id)
+        self.coordinator = coordinator
+        self.station_id = station_id
+        self.entity_description = description
+        self._attr_name = f"{station_id} {description.name}"
+        self._attr_unique_id = unique_id
+        self._attr_icon = "mdi:ev-station"
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, station_id)},
+            manufacturer="InCharge",
+            name=station_id,
+        )
+
+    @property
+    def native_value(self) -> float:
+        """Return the state of the sensor."""
+        result = self.coordinator.data[self.station_id][self.entity_description.key]
+        if self.entity_description.suggested_display_precision is not None:
+            result = round(result, self.entity_description.suggested_display_precision)
+        return result
+
+
+class InChargeDataCoordinator(DataUpdateCoordinator):
+    """The class for handling data retrieval."""
+
+    def __init__(self, hass, incharge_api):
+        """Initialize the InCharge data coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="InCharge DC",
+            update_interval=datetime.timedelta(minutes=5),
+        )
+        self.incharge_api = incharge_api
+
+    async def _async_update_data(self):
+        """Fetch data from InCharge API endpoint."""
+        try:
+            async with async_timeout.timeout(10):
```

Comment:
I think coordinator will handle timeout, I'm not 100% sure though, something worth looking into

Reply:
I'll stick to docs here.